File: src/java/org/voegtle/weatherstation/server/request/ImageUrlParameter.java
package org.voegtle.weatherstation.server.request;

import org.voegtle.weatherstation.server.util.DateUtil;

import javax.servlet.http.HttpServletRequest;

public class ImageUrlParameter extends AbstractUrlParameter {
  private static final String PARAM_OID = "oid";
  private static final String PARAM_ZX = "zx";
  private static final String PARAM_FORMAT = "format";
  private static final String PARAM_SHEET = "sheet";
  private static final String PARAM_CLEAR = "clear";
  private static final String PARAM_REFRESH = "refresh";
  private static final String PARAM_BEGIN = "begin";
  private static final String PARAM_END = "end";

  private final String oid;
  private final String zx;
  private final String format;
  private final boolean refresh;
  private final Integer sheet;
  private boolean clear;
  private final Integer begin;
  private final Integer end;

  public ImageUrlParameter(HttpServletRequest request, DateUtil dateUtil) {
    super(request, dateUtil);
    oid = getUrlParameter(PARAM_OID);
    zx = getUrlParameter(PARAM_ZX);
    format = getUrlParameter(PARAM_FORMAT);
    refresh = getUrlParameterBoolean(PARAM_REFRESH);
    clear = getUrlParameterBoolean(PARAM_CLEAR);
    begin = getUrlParameterInteger(PARAM_BEGIN);
    end = getUrlParameterInteger(PARAM_END);
    sheet = getUrlParameterInteger(PARAM_SHEET, 0);
  }


  public String getOid() {
    return oid;
  }

  public String getZx() {
    return zx;
  }

  public boolean isRefresh() {
    return refresh;
  }

  public boolean isClear() {
    return clear;
  }

  public Integer getBegin() {
    return begin;
  }

  public Integer getEnd() {
    return end;
  }

  public String getFormat() {
    return format;
  }

  public Integer getSheet() {
    return sheet;
  }
}

File: src/main/java/org/voegtle/weatherstation/server/request/ImageUrlParameter.kt
package org.voegtle.weatherstation.server.request

import jakarta.servlet.http.HttpServletRequest
import org.voegtle.weatherstation.server.util.DateUtil

class ImageUrlParameter(request: HttpServletRequest, dateUtil: DateUtil) : AbstractUrlParameter(request, dateUtil) {
  private val PARAM_OID = "oid"
  private val PARAM_ZX = "zx"
  private val PARAM_FORMAT = "format"
  private val PARAM_SHEET = "sheet"
  private val PARAM_CLEAR = "clear"
  private val PARAM_REFRESH = "refresh"
  private val PARAM_BEGIN = "begin"
  private val PARAM_END = "end"

  val oid: String? = getUrlParameter(PARAM_OID)
  val zx: String? = getUrlParameter(PARAM_ZX)
  val format: String? = getUrlParameter(PARAM_FORMAT)
  val isRefresh: Boolean = getUrlParameterBoolean(PARAM_REFRESH)
  val sheet: Int? = getUrlParameterInteger(PARAM_SHEET, 0)
  val isClear: Boolean = getUrlParameterBoolean(PARAM_CLEAR)
  val begin: Int? = getUrlParameterInteger(PARAM_BEGIN)
  val end: Int? = getUrlParameterInteger(PARAM_END)
}
